pkg/certgenerator/v1beta1/generator.go

```diff
@@ -73,7 +73,7 @@ func (c *CertGenerator) Start(ctx context.Context) error {
 }
 
 func (c *CertGenerator) NeedLeaderElection() bool {
-	return true
+	return false
```

Comment:
We must launch the cert-generator on all controllers since the webhook server launches on all controllers even if we set the leanderElection as false.

Reply:
@tenzen-y In that case, we are still going to generate cert and update secret only once, right ?
Since we check if Secret contains data here: https://github.com/kubeflow/katib/blob/5e349feeee84121e732544a3435e471a2a2943a1/pkg/certgenerator/v1beta1/generator.go#L132

Reply:
Yes, that's right. So, I think we shouldn't throw errors in the https://github.com/kubeflow/katib/blob/eac65bf8e31fe4d3a2c4dbb20be30d600c086157/pkg/certgenerator/v1beta1/generator.go#L126-L132.

https://github.com/kubeflow/katib/pull/2185#discussion_r1283746496